scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3086,6 +3087,8 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) {
       case _ => if (hasExplicitMods) syntaxErrorExpected[KwVal]; None
     }
     varOrVarParamMod.foreach(mods += _)
+    if (!mods.has[Mod.ValParam] && !mods.has[Mod.VarParam])
+      rejectMod[Mod.Tracked](mods, "`tracked' modifier can only be used val/var parameters in classes")
```

Comment:
i am generally leaning towards removing validation unless it resolves ambiguity. if the code is correct, parser doesn't need to check; and if it isn't, the compiler will error out.